common/types/layer.go

```diff
@@ -237,3 +237,15 @@ func NewLayer(layerIndex LayerID) *Layer {
 		blocks:  make([]*Block, 0, 3),
 	}
 }
+
+// MinLayer returns the lower non-zero layer.
+func MinLayer(i, j LayerID) LayerID {
```

Comment:
Then I suggest opening an issue to remove `types.MinLayer()` and to refactor the code as it's not really part of bumping to a newer go version.

Reply:
I disagree; this PR also replaces `MayLayer` with `max`. I intentionally left the PR as draft as it was neither urgent nor important and I wanted to come back to it next week 🙂

Reply:
IMHO, the bump it would be best to not mix go version bump with code refactoring at all. This PR should just be an update of go.mod, CI, Dockerfile etc. without any changes in src code, which should happen in a separate PR 🤷

Reply:
No, why update to go 1.21 if we are not using new features. Please don't take over other peoples PRs and redefine their scope 😞

Reply:
This PR is neither important nor urgent... We have a blacklog of 200 issues ready to be worked on 🙂

Reply:
> No, why update to go 1.21 if we are not using new features?

Because it's useful for other reasons too, for example, it would unblock https://github.com/spacemeshos/poet/issues/363. What's the problem with refactoring the remaining parts of the code later?